### ISSUE TITLE
Javascript SDK Information

### ISSUE THREAD

**Issue by Vicken52** [steveseguin/obsninja#687]
Hello @steveseguin, I saw your post on #677 and while I'm not sure what their use case might be, I have some use case that I would like an SDK for.

I'm hoping to develop an overlay wrapper around obsninja for use in RPG / Talk shows with guests on Twitch.
One of the main issues that a lot of people run into when running these shows is overlay management. Weather it's camera placement when someone disconnects or connects or just managing the names and under who's names they are placed.
I would like to be able to use an SDK to connect to an existing room and using the events / information be able to dynamically manage the labels and overlay.

This would probably involve the following requirements from an SDK:
* Being able to listen into events (connecting and disconnecting) to be able to adjust the overlay automatically
* Being able to get a video object for each guest with label and the video html object that can easily be transformed and appended to the correct locations

I believe this would be a powerful tool for content creators allowing them to have producer level overlay management without needing the budget to do so.
My goal is to use this with https://github.com/nodecg/nodecg so that assets for camera borders and general backgrounds can be added and managed dynamically.

Thank you for your time and all the hard work you put into this I'm loving it so far!

**Comment by steveseguin** [comment]
Hi Vicken52, 

Thank  you for that feedback.  

Have you considered using the IFRAME API I've created?  It lets you interact with OBS.Ninja via an API and treat OBS.Ninja video windows as just elements that appear much like video elements.  Supporting a javascript SDK is something I'm aspiring too, but the timeline for when it will be done is unknown.  It will also be much harder for me to support and maintain an SDK versus the IFRAME API.

There's documentation on how to use the IFRAME API; the speedtest I've made is an example of it in action; https://obs.ninja/speedtest

Documentation on it here: https://github.com/steveseguin/obsninja/blob/master/IFRAME.md
